pkg/storagemigration/migrate.go

```diff
@@ -14,6 +14,22 @@ import (
 
 // Migrate migrates the state of the storage from aufs -> overlay2
 func Migrate(root string) error {
+	if logpath, ok := os.LookupEnv("BALENA_MIGRATE_OVERLAY_LOGFILE"); ok {
+		// setup a logrus hook to duplicate logs at <logpath>
+		teardownLogs, err := setupLogs(logpath)
+		if err != nil {
+			return err
+		}
+		// remove the logrus hook
+		defer func() {
+			teardownLogs()
+			if err == nil {
```

Comment:
I assume here you wanted to check for errors during the execution of `Migrate()`, but this `err` refers to the variable defined right above -- and known to be `nil`, so we are removing the logs _always_.

Perhaps best thing here is to use a named return value:

```go
func Migrate(root string) (err error) {
```

Reply:
ouch good catch o/